marklogic/documents.py

```diff
@@ -267,10 +267,56 @@ def _extract_values_from_header(part) -> dict:
     }
 
 
+def multipart_response_to_documents(response: Response) -> list[Document]:
```

Comment:
Extracted this from the `read` method so it can be reused by `search` and possibly for other endpoints that can return a multipart/mixed response for documents.